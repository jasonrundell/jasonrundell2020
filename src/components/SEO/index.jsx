import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'

const Seo = (props) => (
  <Helmet
    htmlAttributes={{
      lang: props.lang,
    }}
    title={props.title}
    meta={[
      {
        name: `description`,
        content: props.description,
      },
      {
        property: `og:title`,
        content: props.title,
      },
      {
        property: `og:description`,
        content: props.description,
      },
      {
        property: `og:type`,
        content: `website`,
      },
      {
        name: `twitter:card`,
        content: `summary`,
      },
      {
        name: `twitter:creator`,
        content: props.author,
      },
      {
        name: `twitter:title`,
        content: props.title,
      },
      {
        name: `twitter:description`,
        content: props.description,
      },
    ].concat(props.meta)}
    bodyAttributes={{
      class: 'theme__vitan',
    }}
  ></Helmet>
)

Seo.defaultProps = {
  lang: `en`,
  meta: [],
  description: ``,
}

Seo.propTypes = {
  description: PropTypes.string,
  meta: PropTypes.arrayOf(PropTypes.object),
  author: PropTypes.string,
  lang: PropTypes.string,
  title: PropTypes.string.isRequired,
}

export default Seo
