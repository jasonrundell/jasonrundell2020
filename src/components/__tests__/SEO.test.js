import React from 'react'
import { create } from 'react-test-renderer'

import SEO from '../SEO'

describe('SEO component', () => {
  test('Matches the snapshot', () => {
    const component = create(
      <SEO
        title="Seo title"
        description="Seo description"
        author="Seo author"
        lang="en"
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
