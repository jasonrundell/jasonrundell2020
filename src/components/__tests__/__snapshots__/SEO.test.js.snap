// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO component Matches the snapshot 1`] = `null`;
